### PR TITLE
Rename /machineDeployments to /nodePools

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/roadmap/issues/1990

This PR renames `/machineDeployments` to `/nodePools`.

### CODE REVIEW

**Comment by primeroz** [comment]
I see this is related to https://github.com/giantswarm/cluster-azure/pull/63

Ideally if we do this we should also add support for rendering either `machineDeployments` or `machinePools` out of the same `nodePools` configuration using a key in the spec like 

```
nodepools
  - name: pool1
    type: machinePool
    location: westeurope
  - name: deployment1
    type: machineDeployment
```

**Comment by marians** [comment]
> Ideally if we do this we should also add support for rendering either machineDeployments or machinePools out of the same nodePools configuration using a key in the spec like

That can definitely be done at some point, ideally by someone else but me.

**Comment by primeroz** [comment]
We now support creating `machineDeployment` and `machinePool` out of the same `nodePool` spec, we need to

* default to `machineDeployment`
* We can make the schema to accept only machineDeployment for now

File: helm/cluster-azure/values.schema.json [hunk 1]
```diff
@@ -273,70 +305,46 @@
...
                     "rootVolumeSizeGB": {
                         "title": "Root volume size (GB)",
                         "type": "integer"
+                    },
+                    "type": {
+                        "enum": [
+                            "machineDeployment",
+                            "machinePool"
+                        ],
+                        "title": "type of NodePool",
```
Comment (marians) [inline helm/cluster-azure/values.schema.json:338]:
```suggestion
                        "title": "Type of node pool",
```

File: helm/cluster-azure/values.schema.json [hunk 2]
```diff
@@ -273,70 +305,46 @@
...
                     },
+                    "replicas": {
+                        "title": "Number of nodes in the MachineDeployment",
+                        "type": "integer"
+                    },
                     "rootVolumeSizeGB": {
                         "title": "Root volume size (GB)",
                         "type": "integer"
+                    },
+                    "type": {
```
Comment (marians) [inline helm/cluster-azure/values.schema.json:333]: I think there should be a default value on the `type` property (probably `machineDeployment` for now).

**Comment by primeroz** [comment]
I had a quick chat with @bavarianbidi and thinks as well that supporting both MD and MP through the `nodePools` key right now is not needed 

I will remove it so that we simplify this change but i will keep the `_machine_pools.tpl` template so we can easily restore it for testing  

the reason why i want to keep the code for MP around is that , when we are ready to switch from MD to MP, we will need to be able to create both to migrate 

* MD Exist for running cluster
* Create MP 
* Cordon and migrate workloads from MD to MP
* delete MP
* get rid of MP support in the code

**Comment by primeroz** [comment]
I removed the support for creating both type of machines from `nodepools` , we will create MD and keep the templated for MP for further testing

```
# [🧰 devenv]  ~/Work/giantswarm/azure/cluster-azure/helm/cluster-azure ☸ glippy (org-multi-project) ( schema-nodepools) 
 λ helm template -f ci/ci-values.yaml . > /dev/shm/render-nodepools.yaml

# [🧰 devenv]  ~/Work/giantswarm/azure/cluster-azure/helm/cluster-azure ☸ glippy (org-multi-project) ( schema-nodepools) 
➜ git co main
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
# [🧰 devenv]  ~/Work/giantswarm/azure/cluster-azure/helm/cluster-azure ☸ glippy (org-multi-project) ( main) 
➜ git pull
Already up to date.
# [🧰 devenv]  ~/Work/giantswarm/azure/cluster-azure/helm/cluster-azure ☸ glippy (org-multi-project) ( main) took 1s 
➜ helm template -f ci/ci-values.yaml . > /dev/shm/render-main.yaml     


 λ diff -u /dev/shm/render-main.yaml /dev/shm/render-nodepools.yaml | wc -l
0
```

**Comment by primeroz** [comment]
@bavarianbidi mind having a look ? this LGTM now but since i worked on it i don't want to approve it myself

**Review by bavarianbidi — APPROVED** [review]

**Comment by marians** [comment]
Just to make sure @primeroz : With this PR, as it stands, we are also removing the /machineDeployments schema. Is this what we want?

**Comment by primeroz** [comment]
and moving it to `/nodePools` yeah it is fine 

`machinePools` code is still supported but not part of the values or the schema

**Comment by marians** [comment]
> and moving it to `/nodePools` yeah it is fine
> 
> `machinePools` code is still supported but not part of the values or the schema

OK, I made a mistake in my question. We are removing schema for `/machinePools` and I wanted to know whether this is actually fine. I understand now that it is fine.

I will improve the changelog entry.

**Review by primeroz — APPROVED** [review]

**Pull request merged**